crates/relayer-cli/src/commands/tx/channel.rs

```diff
@@ -131,8 +146,84 @@ impl Runnable for TxChanOpenInitCmd {
+                height: QueryHeight::Latest,
+            },
+            IncludeProof::No,
+        ) {
+            Ok((client_state, _)) => client_state,
+            Err(e) => Output::error(e).exit(),
+        };
+
+        let mut assembled_hops: Vec<ConnectionHop> = Vec::new();
+
+        assembled_hops.push(ConnectionHop {
+            connection: IdentifiedConnectionEnd::new(
+                self.dst_conn_id.clone(),
+                dst_connection.clone(),
+            ),
+            reference_chain_id: conn_client_state.chain_id().clone(),
+        });
+
+        // Check if connection IDs were provided via --connection-hops, indicating a multi-hop channel
+        if let Some(connnection_ids) = &self.conn_hop_ids {
+            // Retrieve information for each of the remaining hops until the other end of the channel is reached
+            for connection_id in connnection_ids.iter().rev() {
+                // Retrieve the ChainId of the chain referenced by the previous connection hop
+                let chain_id = assembled_hops.last().unwrap().reference_chain_id();
+
+                // Spawn a handle for the chain referenced in the previous hop
+                let chain_handle = match spawn_chain_runtime(&config, &chain_id) {
+                    Ok(handle) => handle,
+                    Err(e) => Output::error(e).exit(),
+                };
+
+                // Retrieve the connection associated with the next hop
+                let hop_connection = match chain_handle.query_connection(
+                    QueryConnectionRequest {
+                        connection_id: connection_id.clone(),
+                        height: QueryHeight::Latest,
+                    },
+                    IncludeProof::No,
+                ) {
+                    Ok((connection, _)) => connection,
+                    Err(e) => Output::error(e).exit(),
+                };
+
+                // Retrieve the state of the client underlying the hop connection
+                let hop_conn_client_state = match chain_handle.query_client_state(
+                    QueryClientStateRequest {
+                        client_id: hop_connection.client_id().clone(),
+                        height: QueryHeight::Latest,
+                    },
+                    IncludeProof::No,
+                ) {
+                    Ok((client_state, _)) => client_state,
+                    Err(e) => Output::error(e).exit(),
+                };
+
+                assembled_hops.push(ConnectionHop {
+                    connection: IdentifiedConnectionEnd::new(
+                        connection_id.clone(),
+                        hop_connection.clone(),
+                    ),
+                    reference_chain_id: hop_conn_client_state.chain_id().clone(),
+                });
+            }
+        }
+
+        // Instantiate ConnectionHops from a vec of ConnectionHop
+        let connection_hops = Some(ConnectionHops::new(assembled_hops));
```

Comment:
Should this be `None` if there are no connection hops?

Reply:
From my understanding, there should always be at least one connection hop. In the case of a single-hop channel, this would be `--dst-connection` pointing to the other end of the channel. Therefore, we always add `--dst-connection` to `connection_hops` and then we process the additional hops if they have been specified. 

From previous manual testing with `gaiad`, the chain requires at least one connection hop to be specified to open the channel. Moreover, in the [ICS33 spec here](https://github.com/cosmos/ibc/blob/main/spec/core/ics-033-multi-hop/README.md?plain=1#L62) it says that a multi-hop packet should be identifiable by the existence of **more than one** connection ID in the `connection_hops` field, implying that single-hop channels have one connection ID in `connection_hops`.

Perhaps this does not need to be an `Option` given that it will always be `Some()` here. At this stage however, having it as an option allows us to pass `None` to `connection_hops` during `Channel` instantiation in other places where we don't have access to the hops yet.

Reply:
As discussed during the call, let's only pass `Some(_)` if there are multiple hops and `None` if there is a single one.